irohad/ametsuchi/impl/postgres_setting_query.cpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/postgres_setting_query.hpp"
+
+#include <boost/lexical_cast.hpp>
+#include "interfaces/common_objects/types.hpp"
+#include "logger/logger.hpp"
+
+using namespace iroha::ametsuchi;
+
+namespace {
+  template <typename T>
+  bool getValueFromDb(soci::session &sql,
+                      const shared_model::interface::types::SettingKeyType &key,
+                      T &destination,
+                      const logger::LoggerPtr &log) {
+    boost::optional<shared_model::interface::types::SettingValueType> value;
+
+    try {
+      sql << "SELECT setting_value FROM setting WHERE setting_key = :key",
+          soci::into(value), soci::use(key, "key");
+    } catch (const std::exception &e) {
+      log->error("Failed to execute query: {}", e.what());
+      return false;
+    }
+
+    if (value) {
+      try {
+        destination = boost::lexical_cast<T>(value.get());
+        return true;
+      } catch (const boost::bad_lexical_cast &e) {
+        log->error("Failed to get value: {}", e.what());
+        return false;
+      }
+    }
+    return false;
+  }
+}  // namespace
+
+PostgresSettingQuery::PostgresSettingQuery(std::unique_ptr<soci::session> sql,
+                                           logger::LoggerPtr log)
+    : psql_(std::move(sql)), sql_(*psql_), log_(std::move(log)) {}
+
+std::shared_ptr<const shared_model::validation::Settings>
+PostgresSettingQuery::get() {
+  std::shared_ptr<shared_model::validation::Settings> kDefaultSettings =
+      std::make_shared<shared_model::validation::Settings>();
+  return update(kDefaultSettings);
+}
+
+std::shared_ptr<const shared_model::validation::Settings>
+PostgresSettingQuery::update(
+    std::shared_ptr<shared_model::validation::Settings> base) {
+  auto get_and_log =
+      [this](const shared_model::interface::types::SettingKeyType &key,
+             auto &destination) {
+        if (getValueFromDb(sql_, key, destination, log_)) {
+          log_->info("Updated value for " + key + ": {}", destination);
+        } else {
+          log_->info("Kept value for " + key + ": {}", destination);
```

Comment:
here no distinction is made between a failure to load a value present in database and a default value usage for a missing value in database. IMO if we fail to load some value from the ledger we should panic and abort.

Reply:
We decide that `SettingQuery` doesn't throw any exceptions
And every error will be processed and replaced by the default value

Reply:
I believe it is wrong and am strongly against it.

Reply:
https://github.com/hyperledger/iroha/pull/133#discussion_r318118991

Reply:
this makes no difference to what I argue about:
> if we fail to load some value from the ledger we should panic and abort.
I think it is necessary because the node must act in accordance with its genesis block, and if it cannot, it should stop operation to prevent inconsistent results.

Reply:
Made that get method returns expected::Result with an error when it is occur

Reply:
Made that get method return expected::Result with an error when it is occurring